// slunce-ui/dist/slunce-ui.js
import { defineComponent as c, computed as a, openBlock as d, createElementBlock as i, normalizeClass as p, unref as u, normalizeStyle as m, renderSlot as f } from "vue";
const _ = {
  name: "sl-button"
}, b = /* @__PURE__ */ c({
  ..._,
  props: {
    type: {
      type: String,
      default: "default"
    },
    disabled: Boolean,
    round: Boolean,
    size: {
      type: String,
      default: "default"
    },
    customColor: {
      type: String,
      default: ""
    },
    leftIcon: String,
    rightIcon: String,
    localing: Boolean
  },
  setup(e) {
    const t = e, o = {
      background: t.customColor,
      border: t.customColor,
      color: "#fff"
    }, s = a(() => [
      t.size == "medium" ? "sl-button-medium" : t.size == "small" ? "sl-button-small" : t.size == "mini" ? "sl-button-mini" : "sl-button",
      t.type ? `sl-button-${t.type}` : "",
      t.disabled ? `sl-button-${t.type ? t.type : "default"}-disabled` : "",
      t.round ? `sl-button-${t.size ? t.size : "default"}-round` : ""
    ]), l = a(() => [
      t.customColor == "" ? {} : t.type == "default" ? {} : o
    ]);
    return (r, z) => (d(), i("button", {
      class: p(["slButton", u(s)]),
      style: m(u(l))
    }, [
      f(r.$slots, "default", {}, void 0, !0)
    ], 6));
  }
});
const y = (e, t) => {
  const o = e.__vccOpts || e;
  for (const [s, l] of t)
    o[s] = l;
  return o;
}, n = /* @__PURE__ */ y(b, [["__scopeId", "data-v-a9940aa4"]]);
n.install = (e) => {
  e.component(n.name, n);
};
const g = [
  n
], S = (e) => {
  g.forEach((t) => e.component(t.name, t));
}, C = {
  install: S
};
export {
  C as default,
  n as slButton
};
