// slunce-ui/dist/slunce-ui.js
import { defineComponent as h, computed as c, openBlock as g, createElementBlock as y, normalizeClass as b, unref as a, normalizeStyle as v, renderSlot as $, createElementVNode as z } from "vue";
const k = {
  name: "sl-button"
}, w = /* @__PURE__ */ h({
  ...k,
  props: {
    type: {
      type: String,
      default: "default"
    },
    disabled: Boolean,
    round: Boolean,
    plain: Boolean,
    size: {
      type: String,
      default: "default"
    },
    color: {
      type: String,
      default: "#000"
    },
    leftIcon: String,
    rightIcon: String,
    localing: Boolean
  },
  setup(e) {
    const t = e, o = (n, m) => {
      let f = [
        parseInt("0x" + n.slice(1, 3)),
        parseInt("0x" + n.slice(3, 5)),
        parseInt("0x" + n.slice(5, 7))
      ], i = [];
      for (var _ = 0; _ < f.length; _++) {
        let r = f[_] + m;
        r < 0 && (r = 0), r > 255 && (r = 255), i.push(r);
      }
      var I = i[0], x = i[1], C = i[2], B = "#" + ((1 << 24) + (I << 16) + (x << 8) + C).toString(16).slice(1);
      return B;
    }, l = {
      "--custom-color": t.color,
      "--custom-color-hover": o(t.color, 40),
      "--custom-color-active": o(t.color, -40),
      "--custom-color-disabled": o(t.color, 80)
    }, s = c(() => [
      t.size == "medium" ? "sl-button-medium" : t.size == "small" ? "sl-button-small" : t.size == "mini" ? "sl-button-mini" : "sl-button sl-button-normal",
      t.type ? `sl-button-${t.type}` : "",
      t.disabled ? "is-disabled" : "",
      t.plain ? "is-plain" : "",
      t.round ? "is-round" : ""
    ]), p = c(() => [
      t.color == "" ? {} : t.type == "default" ? {} : l
    ]);
    return (n, m) => (g(), y("button", {
      class: b(["slButton", a(s)]),
      style: v(a(p))
    }, [
      $(n.$slots, "default", {}, void 0, !0)
    ], 6));
  }
});
const S = (e, t) => {
  const o = e.__vccOpts || e;
  for (const [l, s] of t)
    o[l] = s;
  return o;
}, u = /* @__PURE__ */ S(w, [["__scopeId", "data-v-ac3db428"]]);
u.install = (e) => {
  e.component(u.name, u);
};
const N = ["width", "height"], E = ["xlink:href"], A = {
  name: "sl-icon"
}, D = /* @__PURE__ */ h({
  ...A,
  props: {
    name: {
      type: String,
      default: ""
    },
    color: {
      type: String,
      default: ""
    },
    width: {
      type: [Number, String],
      default: 60
    },
    height: {
      type: [Number, String],
      default: 60
    }
  },
  setup(e) {
    const t = e, o = c(() => `#icon-${t.name}`), l = c(() => `color: ${t.color}`), s = c(() => ["sl-icon", t.name ? `icon-${t.name}` : ""]);
    return (p, n) => (g(), y("svg", {
      class: b(a(s)),
      style: v(a(l)),
      width: t.width,
      height: t.height,
      "aria-hidden": "true"
    }, [
      z("use", {
        "xlink:href": a(o),
        fill: "currentColor"
      }, null, 8, E)
    ], 14, N));
  }
});
const d = /* @__PURE__ */ S(D, [["__scopeId", "data-v-77c97760"]]);
d.install = (e) => {
  e.component(d.name, d);
};
const O = [
  u,
  d
], U = (e) => {
  O.forEach((t) => e.component(t.name, t));
}, j = {
  install: U
};
export {
  j as default,
  u as slButton,
  d as slIcon
};
