// slunce-ui/dist/slunce-ui.js
import { defineComponent as u, computed as a, openBlock as i, createElementBlock as d, normalizeClass as p, unref as c, normalizeStyle as m, renderSlot as _ } from "vue";
const f = {
  name: "sl-button"
}, y = /* @__PURE__ */ u({
  ...f,
  props: {
    type: {
      type: String,
      default: "default"
    },
    disabled: Boolean,
    round: Boolean,
    plain: Boolean,
    size: {
      type: String,
      default: "default"
    },
    color: {
      type: String,
      default: "#000"
    },
    leftIcon: String,
    rightIcon: String,
    localing: Boolean
  },
  setup(e) {
    const t = e, o = {
      "--custom-color": t.color
    }, l = a(() => [
      t.size == "medium" ? "sl-button-medium" : t.size == "small" ? "sl-button-small" : t.size == "mini" ? "sl-button-mini" : "sl-button",
      t.type ? `sl-button-${t.type}` : "",
      t.disabled ? "is-disabled" : "",
      t.plain ? "is-plain" : "",
      t.round ? "is-round" : ""
    ]), s = a(() => [
      t.color == "" ? {} : t.type == "default" ? {} : o
    ]);
    return (r, B) => (i(), d("button", {
      class: p(["slButton", c(l)]),
      style: m(c(s))
    }, [
      _(r.$slots, "default", {}, void 0, !0)
    ], 6));
  }
});
const b = (e, t) => {
  const o = e.__vccOpts || e;
  for (const [l, s] of t)
    o[l] = s;
  return o;
}, n = /* @__PURE__ */ b(y, [["__scopeId", "data-v-4d739fc3"]]);
n.install = (e) => {
  e.component(n.name, n);
};
const g = [
  n
], S = (e) => {
  g.forEach((t) => e.component(t.name, t));
}, z = {
  install: S
};
export {
  z as default,
  n as slButton
};
